### PR TITLE
feat: support agent tokens

### PR DESCRIPTION
**Pull request by martinm82**
First of all, thank you so much for this fantastic Python client for the QuickBuild REST API. The code is so clean and it was a pleasure to implement this feature.

About this PR: It is introducing support for the tokens endpoint which one can use to authorize/unauthorize build agents.

### CODE REVIEW

**Comment by martinm82** [comment]
@pbelskiy the builds are currently failing for PRs coming from forks. This is because of some issue in coveralls and accessing GitHub secrets.
As far as I have seen there is no solution for coveralls. One workaround would be to move to codecov which supports tokenless. See https://github.com/codecov/codecov-action/issues/29#issuecomment-594958141. Another one would be to skip coverage on PRs from forks but I don't believe this is the right way or check https://github.blog/2020-08-03-github-actions-improvements-for-fork-and-pull-request-workflows/. Seems that there might be a setting to make coveralls work with tweaking some settings.

File: quickbuild/endpoints/tokens.py [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+        Args:
+            agent_ip (str): The build agent IP address.
+            agent_port (int): The build agent port.
+
+        Returns:
+            str: identifier of the newly created token for the build agent
+        """
+        response = self.quickbuild._request(
+            'GET',
+            'tokens/authorize?ip={}&port={}'.format(agent_ip, agent_port)
```
Comment (pbelskiy) [inline quickbuild/endpoints/tokens.py]: It's more correct to use params argument for request:

params=dict(ip=agent_ip, port=agent_port)

File: quickbuild/endpoints/tokens.py [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+        Args:
+            agent_ip (str): The build agent IP address.
+            agent_port (int): The build agent port.
+
+        Returns:
+            str: identifier of the removed token representing the build agent.
+        """
+        response = self.quickbuild._request(
+            'GET',
+            'tokens/unauthorize?ip={}&port={}'.format(agent_ip, agent_port)
```
Comment (pbelskiy) [inline quickbuild/endpoints/tokens.py]: same

File: quickbuild/endpoints/tokens.py [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+        )
+
+        return response
+
+    def get_token_and_agent_details(self, agent_address: str) -> List[dict]:
```
Comment (pbelskiy) [inline quickbuild/endpoints/tokens.py]: I would rename it to get() method with optional parameter agent_address as other endpoints do

File: quickbuild/endpoints/tokens.py [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+        """
+        Get token value and latest used information of agents.
+
+        Args:
+            agent_address (str): Build agent address, eg. my-agent:8811.
```
Comment (pbelskiy) [inline quickbuild/endpoints/tokens.py]: it's optional parameter from documentation

File: quickbuild/endpoints/tokens.py [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+            root = xmltodict.parse(response)
+
+            tokens = []
+            if root['list'] is not None:
+                tokens = root['list']['com.pmease.quickbuild.model.Token']
+                if isinstance(tokens, list) is False:
+                    tokens = [tokens]
+            return tokens
+
+        agent_address_query_string = '?address={}'.format(agent_address) if agent_address else ''
```
Comment (pbelskiy) [inline quickbuild/endpoints/tokens.py]: please use params argument for request

**Review by pbelskiy — APPROVED** [review]
Thanks for PR! Looks good to me except few things I commented. 

Also please add to docs/index.rst

```
Tokens
~~~~~~

.. autoclass:: quickbuild.endpoints.tokens.Tokens
    :members:
```

File: quickbuild/endpoints/tokens.py [hunk 6]
```diff
@@ -68,10 +70,11 @@ def callback(response: str) -> List[dict]:
                     tokens = [tokens]
             return tokens
 
-        agent_address_query_string = '?address={}'.format(agent_address) if agent_address else ''
+        params_agent_address = dict(address=agent_address) if agent_address else []
```
Comment (pbelskiy) [inline quickbuild/endpoints/tokens.py]: else {}

File: quickbuild/endpoints/tokens.py [hunk 7]
```diff
@@ -49,13 +49,14 @@ def unauthorize(self, agent_ip: str, agent_port: int = 8811) -> str:
 
         return response
 
-    def get_token_and_agent_details(self, agent_address: str) -> List[dict]:
+    def get(self, agent_address: str = None) -> List[dict]:
```
Comment (pbelskiy) [inline quickbuild/endpoints/tokens.py]: Optional[str]

File: docs/index.rst [hunk 8]
```diff
@@ -27,6 +27,12 @@ Users
 .. autoclass:: quickbuild.endpoints.users.Users
     :members:
 
+Tokens
+~~~~~
```
Comment (pbelskiy) [inline docs/index.rst:31]: CI failed because to short, you need to add additional `~`

File: quickbuild/core.py [hunk 9]
```diff
@@ -4,6 +4,7 @@
 from typing import Any, Callable, NamedTuple, Optional
 
 from quickbuild.endpoints.builds import Builds
+from quickbuild.endpoints.tokens import Tokens
```
Comment (pbelskiy) [inline quickbuild/core.py:7]: it would be nice if you sort endpoints import and use same sequence in init

File: quickbuild/endpoints/tokens.py [hunk 10]
```diff
@@ -0,0 +1,81 @@
...
+        """
+        response = self.quickbuild._request(
+            'GET',
+            'tokens/unauthorize',
+            params=dict(ip=agent_ip, port=agent_port),
+        )
+
+        return response
+
+    def get(self, agent_address: str = None) -> List[dict]:
```
Comment (pbelskiy) [inline quickbuild/endpoints/tokens.py]: please move get to top of class

**Review by pbelskiy — APPROVED** [review]

**Comment by pbelskiy** [comment]
It's okay now, but your new tests fails on CI, you also can test it locally using tox, also please don't forget to rebase before submit, because yesterday I've added `audits` endpoints

**Review by pbelskiy — APPROVED** [review]

**Comment by pbelskiy** [comment]
@martinm82 released under 0.4.0 version

**Comment by martinm82** [comment]
@pbelskiy I think one of the tests was still failing. Sorry, I was recently fully occupied with work and could not check further on this PR

**Comment by pbelskiy** [comment]
@martinm82 Yes, but I've fixed it yesterday

**Pull request merged**